test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -1133,6 +1149,106 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, IDPDiscoveryForm, B
         });
       });
     });
-
+    Expect.describe('Additional Auth Button', function () {
+      itp('does not show the divider and buttons if settings.customButtons is not set', function () {
+        return setup().then(function (test) {
+          expect(test.form.authDivider().length).toBe(0);
+          expect(test.form.additionalAuthButton().length).toBe(0);
+        });
+      });
+      itp('show the divider and buttons if settings.customButtons is not empty', function () {
+        return setupAdditionalAuthButton().then(function (test) {
+          expect(test.form.authDivider().length).toBe(1);
+          expect(test.form.additionalAuthButton().length).toBe(1);
+        });
+      });
+      itp('sets text with property passed', function () {
+        return setupAdditionalAuthButton().then(function(test){
+          expect(test.form.additionalAuthButton().text()).toEqual('test text');
+        });
+      });
+      itp('sets class with property passed', function () {
+        return setupAdditionalAuthButton().then(function(test){
+          expect(test.form.additionalAuthButton().hasClass('test-class')).toBe(true);
+        });
+      });
+      itp('clickHandler is called when button is clicked', function () {
+        return setupAdditionalAuthButton().then(function(test){
+          expect(test.form.additionalAuthButton().hasClass('new-class')).toBe(false);
+          test.form.additionalAuthButton().click();
+          expect(test.form.additionalAuthButton().hasClass('new-class')).toBe(true);
+        });
+      });
+      itp('displays social auth and custom buttons', function () {
+        var settings = {
```

Comment:
Can we have a function for this (and the other tests) as well, similar to `setupAdditionalAuthButton`?